### PR TITLE
refactor: Update some snmpwalks for ports polling to improve speed

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

I'll do some inline comments on this one but for me it drops port polling on Arista down from 160 -> 80 seconds. It also helps our Cisco switches, with about 800 devices I've dropped 20 seconds on each distributed poller.

### CODE REVIEW

File: includes/polling/ports.inc.php [hunk 1]
```diff
@@ -66,23 +66,6 @@
     'ifOutMulticastPkts',
 );
 
-// From above for DB
-$etherlike_oids = array(
```
Comment (laf) [inline includes/polling/ports.inc.php]: We appear to not use these at all.

File: includes/polling/ports.inc.php [hunk 2]
```diff
@@ -224,11 +210,11 @@
     }
 }
 
-if ($config['enable_ports_etherlike']) {
-    echo 'dot3Stats ';
-    $port_stats = snmpwalk_cache_oid($device, 'dot3StatsEntry', $port_stats, 'EtherLike-MIB');
```
Comment (laf) [inline includes/polling/ports.inc.php]: We don't use any of this table.

File: includes/polling/ports.inc.php [hunk 3]
```diff
@@ -224,11 +210,11 @@
...
-} elseif ($device['os'] != 'asa') {
+if ($device['os'] != 'asa') {
     echo 'dot3StatsDuplexStatus';
+    if ($config['enable_ports_poe'] || $config['enable_ports_etherlike']) {
+        $port_stats = snmpwalk_cache_oid($device, 'dot3StatsIndex', $port_stats, 'EtherLike-MIB');
```
Comment (laf) [inline includes/polling/ports.inc.php]: We use only this + dot3StatsDuplexStatus within poe and etherlike port polling files so only look those up.

File: includes/polling/ports.inc.php [hunk 4]
```diff
@@ -265,13 +251,17 @@
...
 }
 
-// FIXME This probably needs re-enabled. We need to clear these things when they get unset, too.
-// foreach ($etherlike_oids as $oid) { $port_stats = snmpwalk_cache_oid($device, $oid, $port_stats, "EtherLike-MIB"); }
-// foreach ($cisco_oids as $oid)     { $port_stats = snmpwalk_cache_oid($device, $oid, $port_stats, "OLD-CISCO-INTERFACES-MIB"); }
-// foreach ($pagp_oids as $oid)      { $port_stats = snmpwalk_cache_oid($device, $oid, $port_stats, "CISCO-PAGP-MIB"); }
 if ($device['os_group'] == 'cisco' && $device['os'] != 'asa') {
     foreach ($pagp_oids as $oid) {
-        $port_stats = snmpwalk_cache_oid($device, $oid, $port_stats, 'CISCO-PAGP-MIB');
+        $pagp_port_stats = snmpwalk_cache_oid($device, $oid, array(), 'CISCO-PAGP-MIB');
```
Comment (laf) [inline includes/polling/ports.inc.php]: Later on we only use the check for pagpOperationMode so we poll that first, check it has data and if so apply it to $port_stats and then poll the rest of pagp data.

File: includes/polling/ports.inc.php [hunk 5]
```diff
@@ -280,7 +270,7 @@
     $port_stats = snmpwalk_cache_oid($device, 'vlanTrunkPortEncapsulationOperType', $port_stats, 'CISCO-VTP-MIB');
     $port_stats = snmpwalk_cache_oid($device, 'vlanTrunkPortNativeVlan', $port_stats, 'CISCO-VTP-MIB');
 } elseif ($device['os'] != 'asa') {
-    $port_stats = snmpwalk_cache_oid($device, 'dot1qPortVlanTable', $port_stats, 'Q-BRIDGE-MIB');
+    $port_stats = snmpwalk_cache_oid($device, 'dot1qPvid', $port_stats, 'Q-BRIDGE-MIB');
```
Comment (laf) [inline includes/polling/ports.inc.php]: We only used dot1qPvid out of the entire table!

**Comment by laf** [comment]
Comments on both old and new changes in: https://github.com/librenms/librenms/pull/6341/files

**Comment by murrant** [comment]
@laf Looks nice!  Not sure when I will have time to test...

**Comment by laf** [comment]
You can have faith :)

**Comment by murrant** [comment]
I have faith, but for some portions of the code I require myself to test before approving ;)

**Comment by laf** [comment]
Wimp :D

**Pull request merged**